module/CMakeLists.txt

```diff
@@ -245,73 +268,59 @@ if(DEAL_II_WITH_CXX20_MODULE)
       set(build_camelcase "Release")
     endif()
 
-    # Define the library. Compile it with the usual flags we also
-    # use for all other targets, but do set the DEAL_II_BUILDING_CXX20_MODULE
+    # Define the library. Compile it with the usual flags we also use for
+    # all other targets, but do set the DEAL_II_BUILDING_CXX20_MODULE
     # preprocessor variable for a small number of cases where we need to
-    # work around things that work differently between module and non-module
-    # builds.
-    #
-    # Note that we call the preprocessor variable
-    # ..._BUILDING_... instead of the overarching
-    # ..._WITH_... variable because we build the same set of files
-    # twice, but only one of these is build *built* for modules,
-    # whereas both sets are being build in a set-up in which we have
-    # modules ("with").
+    # work around things that work differently between module and
+    # non-module builds.
     add_library(${DEAL_II_TARGET_NAME}_module_${build_lowercase} SHARED)
     populate_target_properties(${DEAL_II_TARGET_NAME}_module_${build_lowercase} ${build})
     target_compile_definitions(${DEAL_II_TARGET_NAME}_module_${build_lowercase}
       PRIVATE "DEAL_II_BUILDING_CXX20_MODULE"
       )
 
+    #
     # Say what the source files for the library are. These are all of
     # the interface and implementation partition units we have created
     # above (including the ones that wrap external libraries), plus
-    # the primary interface unit. Some of these will have to be
-    # installed -- specifically, the interface and primary interface
-    # units -- and so need to be PUBLIC.  On the other hand,
-    # downstream projects do not need access to our implementation,
-    # and consequently the implementation units can be marked as
-    # PRIVATE. Finally, the macros.h header is also a PUBLIC source
-    # that needs to be installed.
+    # the primary interface unit.
     #
-    # TODO: It should not be necessary to list BASE_DIRS here. The module
-    # files are create in the build directory, and are all listed as absolute
-    # paths, not relative to a base dir. But cmake still complains when one
-    # omits the BASE_DIRS part, even though there is no reason for it to
-    # check whether these files can be found in the implicit BASE_DIRS (namely,
-    # CMAKE_CURRENT_SOURCE_DIR) because absolute paths to these files are
-    # provided.
+
     target_sources(${DEAL_II_TARGET_NAME}_module_${build_lowercase}
       PUBLIC
-      FILE_SET ${DEAL_II_TARGET_NAME}_module_${build_lowercase}_sources
+      FILE_SET module_interface_units
       TYPE CXX_MODULES
       FILES
         ${_interface_module_partition_units}
         ${_primary_module_interface_unit}
-      BASE_DIRS
-        ${CMAKE_CURRENT_SOURCE_DIR}
-        ${CMAKE_CURRENT_BINARY_DIR}
+      BASE_DIRS "${CMAKE_CURRENT_BINARY_DIR}/interface_module_partitions"
+      )
```

Comment:
This is interesting here. You are declaring the source files for either the debug or release library, but you re-use the same `FILE_SET` name for both. Does this work?

Ah, see, my comment below.

Reply:
Yup! I tried by hand to convince myself that this is indeed true :smile: